h2o-genmodel/src/main/java/hex/genmodel/ModelMojoReader.java

```diff
@@ -240,6 +243,64 @@ else if (line.equals("[columns]")) {
     return domains;
   }
 
+  Map<String, Map<String, int[]>> parseTargetEncodingMap(BufferedReader source) throws IOException {
+    Map<String, Map<String, int[]>> encodingMap = new HashMap<>();
+    Map<String, int[]> encodingsForColumn = null;
+    String sectionName = null;
+    try {
+      String line;
+
+      while (true) {
+        line = source.readLine();
+        if (line == null) { // EOF
+          encodingMap.put(sectionName, encodingsForColumn);
+          break;
+        }
+        line = line.trim();
+        if(sectionName == null) {
+          sectionName = matchNewSection(line);
+          encodingsForColumn = new HashMap<>();
+        } else { 
+          String matchResult = matchNewSection(line);
+          if(matchResult!= null) {
+            encodingMap.put(sectionName, encodingsForColumn);
+            encodingsForColumn = new HashMap<>();
+            sectionName = matchResult;
+            continue;
+          }
+          
+          String[] res = line.split("\\s*=\\s*", 2);
+          int[] numAndDenom = processNumeratorAndDenominator(res[1].split(" "));
+          encodingsForColumn.put(res[0], numAndDenom);
+        }
+      }
+      source.close();
```

Comment:
Use try-with-resources block, avoids manual closing on many places.

Reply:
@Pscheidl I was actually following the logic that we had in `parseModelInfo` method in a given class. And I'm actually using try-with-resources statement here.... but first `source.close()` is inside the `try` block. I need an explanation why we did the same it in `parseModelInfo` above. cc @michalkurka